controllers/controllers/workloads/cftask_controller.go

```diff
@@ -249,6 +253,37 @@ func (r *CFTaskReconciler) ensureInitialized(ctx context.Context, cfTask *korifi
 			Reason:  "taskInitialized",
 			Message: "taskInitialized",
 		})
+
+	}
+
+	return nil
+}
+
+func (r *CFTaskReconciler) handleCancelation(ctx context.Context, cfTask *korifiv1alpha1.CFTask) error {
+	eiriniTask := &eiriniv1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cfTask.Name,
+			Namespace: cfTask.Namespace,
+		},
+	}
+	err := r.k8sClient.Delete(ctx, eiriniTask)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		r.logger.Info("error-deleting-eirini-task", "error", err)
+		return err
+	}
+
+	meta.SetStatusCondition(&cfTask.Status.Conditions, metav1.Condition{
+		Type:   korifiv1alpha1.TaskCanceledConditionType,
+		Status: metav1.ConditionTrue,
+		Reason: "task_canceled",
+	})
+
+	if meta.FindStatusCondition(cfTask.Status.Conditions, korifiv1alpha1.TaskStartedConditionType) != nil {
+		meta.SetStatusCondition(&cfTask.Status.Conditions, metav1.Condition{
```

Comment:
I am a bit worried that the status in the cache might not be up to date and miss the `Started` condition being available, hence `Failed` would be never set. Is this something we need to worry about?